platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/DefaultConvention.java

```diff
@@ -315,7 +316,7 @@ public void propertyMissing(String name, Object value) {
         }
 
         @Override
-        public DynamicInvokeResult tryInvokeMethod(String name, Object... args) {
+        public DynamicInvokeResult tryInvokeMethod(String name, @Nullable Object... args) {
```

Comment:
I'm not sure what the best resolution here is, but this is going to cause problems with #24767. When the new `Nullable` annotation is a `TYPE_USE` instead of a `PARAMETER` target, this will be "array of nullable `Object`" instead of "nullable array of `Object`". I'm not sure if it would be better to just start using JSpecify in a limited form here, or if we should just leave it as-is and try to catch this when the JSpecify work happens.

Reply:
> [in JSpecify] this will be "array of nullable Object" instead of "nullable array of Object"

This is exactly the intention of the change in this PR -- to denote an "non-nullable array of nullable `Object`". To put it more specifically, the `tryInvokeMethod` expects an array of arguments, the array is never null, but each argument can be null.

As far as I was able to gather, this is the correct interpretation of an annotation with a vararg syntax. Similarly, to how `@Nullable Object[]` means "an array of nullable `Object`". Intellij does not seem to get it right, but that's a different story.

Moreover, if this is not the case, we already have a couple of places in the code that use `@Nullable Type...` syntax, so they would have to be fixed as well.

I would stick to the `@Nullable Object...` syntax, and deal with JSpecify semantics later, if needed at all.

Reply:
> this is the correct interpretation of an annotation with a vararg syntax. 

It varies between `TYPE_USE` and `PARAMETER`, `PARAMETER` is always interpreted as applying to the type of the whole parameter, whereas `TYPE_USE` binds to the inner type of the array. So IntelliJ is actually correct since this is a `PARAMETER` annotation, it can only ever represent "nullable array of `Object`", and not "array of nullable `Object`". So you might consider just removing the annotation here, since it's incorrect.

Reply:
You are right, I forgot that annotation target also affects this. So basically, the current setup that we have is incapable of expressing what we need for vararg case, which is an "array of nullable values". Though, I wouldn't want to mix in JSpecify before we commit to it on a higher level. Which, I guess, will happen as soon as it reaches 1.0.

This leaves the choice of either not putting a nullable annotation for varargs at all, or putting it there, at the cost of it not being entirely correct. Between these two, **I would prefer to put the annotation to draw attention to this**.

If we do not annotate (like you suggest), then package-level `NonNullApi` would make it seem like everything is non-nullable. This creates a false sense of security and increases the chance of not handling the null-case of the values.

The main point that we draw developer's attention to the nullability aspect for this edge case. The inconsistency caused by the annotation target being `PARAMETER` can be addressed in internal docs.